include/deal.II/grid/grid_tools.h

```diff
@@ -180,6 +180,10 @@ namespace GridTools
    * some ghost cells may not. This means that computations like
    * KellyErrorEstimator may give wrong answers.
    *
+   * @note Hanging nodes are handled under the assumption of isotropically 
+   * refineed grids. While there are no problems for `dim==2`, this is becomes
+   * troublesome for `dim==3`. The use of this function for anisotropically
+   * refined grids in `dim==3` is thus not implemented and throws an error.
```

Comment:
```suggestion
   * @note Hanging nodes are handled under the assumption of isotropically 
   * refined grids. The use of this function for anisotropically
   * refined grids in `dim==3` is thus not implemented and throws an error.
```

You added the Assert only for the two-dimensional case, though.

Reply:
My bad. You also just made me realize that checking for the refinement case of the cell itself is non-sensical: We are working on the freshly refined mesh and iterate over active cells, thus the included assertion is always given. 

I think the original programmer had isotropically refined faces in mind. With `dim==2` an anisotropically refined cell yields by definition isotropically refined faces, but for `dim==3` anisotropic refinement may yield anisotropic refined faces. The Assertion has thus changed from `cell->refinement_case()` to `cell->face(face)->refinement_case()`.